core/modules/file/file.install

```diff
@@ -713,3 +852,106 @@ function file_update_1002() {
+  $config->set('protect_repeated_render', update_variable_get('file_entity_protect_repeated_render', TRUE));
+  $config->save();
+
+  // Remove converted variables.
+  variable_del('file_entity_max_filesize');
+  variable_del('file_entity_default_allowed_extensions');
+  variable_del('file_entity_default_file_directory');
+  variable_del('file_entity_allow_insecure_download');
+  variable_del('file_entity_file_upload_wizard_skip_file_type');
+  variable_del('file_entity_file_upload_wizard_skip_scheme');
+  variable_del('file_entity_file_upload_wizard_skip_fields');
+  variable_del('file_entity_protect_repeated_render');
+
+  // Delete unused variables.
+  variable_del('file_entity_alt');
+  variable_del('file_entity_title');
+}
+
+/**
+ * Add a type column to the file_managed table. Add the file_metadata table.
+ */
+function file_update_1005() {
+  if (!db_field_exists('file_managed', 'type')) {
+    $spec = array(
+      'description' => 'The type of this file.',
+      'type' => 'varchar',
+      'length' => 50,
+      'not null' => TRUE,
+      // If the FILE_TYPE_NONE constant ever changes, then change the value here
+      // too, and add an update function to deal with existing records. The
+      // constant isn't used here, because there may be cases where this function
+      // runs without the module file loaded.
+      'default' => 'undefined',
+    );
+    $indexes_new = array('indexes' => array('file_type' => array('type')));
+    db_add_field('file_managed', 'type', $spec, $indexes_new);
+  }
+
+  if (!db_table_exists('file_metadata')) {
+    $schema = array(
+      'description' => 'Stores file metadata in a key/value store.',
+      'fields' => array(
+        'fid' => array(
+          'description' => 'The {file_managed}.fid of the metadata.',
+          'type' => 'int',
+          'unsigned' => TRUE,
+          'not null' => TRUE,
+          'default' => 0,
+        ),
+        'name' => array(
+          'description' => "The name of the metadata (e.g. 'width').",
+          'type' => 'varchar',
+          'length' => '255',
+          'not null' => TRUE,
+        ),
+        'value' => array(
+          'description' => "The value of the metadata (e.g. '200px').",
+          'type' => 'blob',
+          'not null' => FALSE,
+          'size' => 'big',
+          'serialize' => TRUE,
+        ),
+      ),
+      'primary key' => array('fid', 'name'),
+      'foreign keys' => array(
+        'file_managed' => array(
+          'table' => 'file_managed',
+          'columns' => array('fid' => 'fid'),
+        ),
+      ),
+    );
+    db_create_table('file_metadata', $schema);
+  }
+}
+
+/**
+ * Add the default file types for existing Backdrop sites.
+ */
+function file_update_1006() {
+  $default_file_types = array('audio', 'document', 'image', 'video');
+  foreach ($default_file_types as $type) {
+    if (!file_type_get_type($type)) {
+      config_install_default_config('file', 'file.type.' . $type);
```

Comment:
We can't assume that the default config will always be the same. Similar to when we have to make DB changes or views, you have to duplicate or write the whole config manually here.

Reply:
reworked.